src/Domain/Entities/Product/ProductValidator.cs

```diff
@@ -0,0 +1,30 @@
+﻿using CleanCompanyName.DDDMicroservice.Domain.Common.Validators;
+using CleanCompanyName.DDDMicroservice.Domain.Entities.Product.ValueObject;
+using FluentValidation;
+
+namespace CleanCompanyName.DDDMicroservice.Domain.Entities.Product;
+
+public class ProductValidator : AbstractValidator<Product>
+{
+    public ProductValidator()
+    {
+        RuleFor(product => product.Id)
+            .NotEmpty();
+
+        RuleFor(product => product.Sku)
+            .NotEmpty()
+            .WithCustomErrorCode(this)
```

Comment:
Here we have an example of how to let our custom validator extension write our code error.